Sources/TuistKit/Services/InitService.swift

```diff
@@ -1,266 +1,207 @@
-    ) async throws {
-        let platform = try self.platform(platform)
-        let path = try self.path(path)
-        let name = try self.name(name, path: path)
-        let templateName = templateName ?? "default"
-        try await verifyDirectoryIsEmpty(path: path)
+        \(tuistSwiftLine)
+        """
+        let tuistSwiftFilePath = projectDirectory.appending(component: "Tuist.swift")
+        if try await fileSystem.exists(tuistSwiftFilePath) {
+            try await fileSystem.remove(tuistSwiftFilePath)
+        }
+        try await fileSystem.writeText(tuistSwiftFileContent, at: tuistSwiftFilePath)
+
+        ServiceContext.current?.alerts?.success(.alert("You are all set to explore the Tuist universe", nextSteps: [
+            "Accelerate your builds with the \(.link(title: "cache", href: "https://docs.tuist.dev/en/guides/develop/cache"))",
+            "Accelerate your test runs with \(.link(title: "selective testing", href: "https://docs.tuist.dev/en/guides/develop/selective-testing"))",
+            "Accelerate your Swift package resolution with \(.link(title: "the registry", href: "https://docs.tuist.dev/en/guides/develop/registry"))",
+            "Share your app easily with \(.link(title: "previews", href: "https://docs.tuist.dev/en/guides/share/previews"))",
+        ]))
+    }
 
-        if templateName.isGitURL {
-            try await templateGitLoader.loadTemplate(from: templateName, closure: { template in
-                let parsedAttributes = try self.parseAttributes(
+    private func createGeneratedProject(at directory: AbsolutePath, name: String, platform: String) async throws -> AbsolutePath {
+        let projectDirectory = directory.appending(component: name)
+        try await ServiceContext.current?.ui?.progressStep(
+            message: "Creating generated project",
+            successMessage: "Generated project created",
+            errorMessage: "Failed to create generated project",
+            showSpinner: true,
+            task: { _ in
+
+                if !(try await fileSystem.exists(projectDirectory)) {
+                    try await fileSystem.makeDirectory(at: projectDirectory)
+                }
+                try await startGeneratedProjectService.run(
                     name: name,
                     platform: platform,
-                    tuistVersion: Constants.version,
-                    requiredTemplateOptions: requiredTemplateOptions,
-                    optionalTemplateOptions: optionalTemplateOptions,
-                    template: template
-                )
-
-                try await self.templateGenerator.generate(
-                    template: template,
-                    to: path,
-                    attributes: parsedAttributes
+                    path: projectDirectory.pathString,
+                    templateName: "default",
+                    requiredTemplateOptions: [:],
+                    optionalTemplateOptions: [:]
```

Comment:
All of parameters can be removed now in favor of the service specifying these things itself.